src/services/item/plugins/app/appAction/repository.ts

```diff
@@ -1,77 +1,81 @@
-import { In } from 'typeorm';
+import { EntityManager, In } from 'typeorm';
 
 import { ResultOf } from '@graasp/sdk';
 
-import { AppDataSource } from '../../../../../plugins/datasource';
-import { MemberIdentifierNotFound } from '../../../../itemLogin/errors';
+import { MutableRepository } from '../../../../../repositories/MutableRepository';
+import { DEFAULT_PRIMARY_KEY } from '../../../../../repositories/const';
+import { IllegalArgumentException } from '../../../../../repositories/errors';
 import { appActionSchema } from '../../../../member/plugins/export-data/schemas/schemas';
 import { schemaToSelectMapper } from '../../../../member/plugins/export-data/utils/selection.utils';
 import { mapById } from '../../../../utils';
 import { ManyItemsGetFilter, SingleItemGetFilter } from '../interfaces/request';
 import { AppAction } from './appAction';
 import { InputAppAction } from './interfaces/app-action';
 
-export const AppActionRepository = AppDataSource.getRepository(AppAction).extend({
-  async post(itemId: string, memberId: string, body: Partial<InputAppAction>): Promise<AppAction> {
-    const created = await this.insert({
+type PostAppActionBody = Partial<InputAppAction> & { itemId: string; memberId: string };
+type PatchAppActionBody = Partial<Omit<AppAction, 'item'>>;
```

Comment:
As we discussed, `Post` and `Patch` are more HTTP related. Should we rename theses to
```suggestion
type CreateAppActionBody = Partial<InputAppAction> & { itemId: string; memberId: string };
type UpdateAppActionBody = Partial<Omit<AppAction, 'item'>>;
```